src/profile-logic/process-profile.js

```diff
@@ -907,7 +907,7 @@ function _processMarkerPayload(
       // Coerce the payload into a MarkerPayload. This doesn't really provide
       // any more type safety, but it shows the intent of going from an object
       // without much type safety, to a specific type definition.
-      return (payload: MarkerPayload);
+      return (payload: any);
```

Comment:
The return type for this function is still `MarkerPayload | null` so we don't lost the type safety by using `any` here.

Reply:
Hmm, but do you know why we can't write `MarkerPayload` here anymore? It looks a bit suspicious. Is it because the payload is not `any` anymore? (also see my comment about why that is not correct)

Also a nit: I think we need to update the comment above of this as we are not coercing it to `MarkerPayload` anymore. But I think we will be able to coerce it again after making the payload `any` again.

Reply:
> Is it because the payload is not any anymore? (also see my comment about why that is not correct)

Yep, exactly. But because the function return value is `MarkerPayload` it will still be coerced to `MarkerPayload` for the users of this function in the end. I can be more explicit here and coerce to MarkerPayload explicitely if you prefer.